File: backend/src/main/java/com/military/doc/modules/document/entity/DocEffectiveBaseline.java
package com.military.doc.modules.document.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("doc_effective_baseline")
public class DocEffectiveBaseline {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;
    private Long stageId;
    private Long docFileId;
    private Long effectiveVersionId;
    private Long finalVersionId;
    private Long confirmedBy;
    private LocalDateTime confirmedAt;
    private String baselineStatus;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/doc-effective-baseline.ts
import api from './index'

export interface DocEffectiveBaselineItem {
  id?: number
  projectId: number
  stageId: number
  docFileId: number
  effectiveVersionId: number
  finalVersionId: number
  confirmedBy: number
  confirmedAt: string
  baselineStatus: string
}

export function getDocEffectiveBaselines(projectId?: number) {
  return api.get('/doc-baselines', { params: { projectId } })
}

export function createDocEffectiveBaseline(data: DocEffectiveBaselineItem) {
  return api.post('/doc-baselines', data)
}

export function updateDocEffectiveBaseline(id: number, data: DocEffectiveBaselineItem) {
  return api.put(`/doc-baselines/${id}`, data)
}

export function deleteDocEffectiveBaseline(id: number) {
  return api.delete(`/doc-baselines/${id}`)
}
